src/firefly/java/edu/caltech/ipac/firefly/server/query/AsyncTapQuery.java

```diff
@@ -55,6 +58,19 @@ public HttpServiceInput createInput(TableServerRequest request) throws DataAcces
             try {
                 inputs.setParam("UPLOAD",uploadTable+","+"param:temp.xml");
                 DataGroup dg= TableUtil.readAnyFormat(ServerContext.convertToFile(request.getParam("UPLOAD")));
+                if (request.containsParam("UPLOAD_COLUMNS")) { //filter dg columns down to the ones the user has selected
+                    DataType[] data = dg.getDataDefinitions();
+                    List<String> colNames = new ArrayList<>();
+                    for (DataType entry : data) {
+                        colNames.add(entry.getKeyName());
+                    }
+                    String uploadCols = request.getParam("UPLOAD_COLUMNS");
+                    String[] cols = uploadCols.split(",");
+                    for (String key : colNames) {
+                        boolean contains = Arrays.stream(cols).anyMatch(key::equals);
+                        if (!contains) dg.removeDataDefinition(key);
+                    }
+                }
```

Comment:
This makes sense.  There's a few minor things to check..
- UPLOAD_COLUMNS is empty string
- colNames contains column with commas in the name
- check the 'saved' file to make sure `dg.removeDataDefinition` worked.

code sample ( rebase `dev` to use `TableUtil.splitCols`)
```java
String uploadCols = request.getParam("UPLOAD_COLUMNS");
if (!StringUtils.isEmpty(uploadCols)) {                                     //filter dg columns down to the ones the user has selected
    List<String> cols = Arrays.stream(TableUtil.splitCols(uploadCols))      // use this to allow commas in double-quotes
                    .map(cname -> cname.trim().replaceAll("^\"|\"$", ""))  // remove double-quotes if exists
                    .toList();

    Arrays.asList(dg.getDataDefinitions()).forEach(c -> {
        if (!cols.contains(c.getKeyName())) dg.removeDataDefinition(c.getKeyName());
    });
}
```